src/main/java/appeng/api/storage/data/IItemList.java

```diff
@@ -70,10 +71,14 @@ public interface IItemList<StackType extends IAEStack> extends IItemContainer<St
     void resetStatus();
 
     default StackType[] toArray(StackType[] zeroSizedArray) {
-        ArrayList<StackType> output = new ArrayList<>(size());
+        int prevSize = size();
+        StackType[] output = (StackType[]) Array.newInstance(zeroSizedArray.getClass().getComponentType(), prevSize);
+
+        int i = 0;
         for (StackType stack : this) {
```

Comment:
any reason not to turn this into c style array access instead of a foreach?

Reply:
I'm not sure what you mean, the reason for this change is explained in the commit [`094dcea` (#525)](https://github.com/GTNewHorizons/Applied-Energistics-2-Unofficial/pull/525/commits/094dcea120006414a3e54ada60abfd1bd1e60e4b)